calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -198,13 +198,17 @@ func (rh resourceHelper) Update(ctx context.Context, client client.Interface, re
 	// If the resourceVersion is specified then we use it to try and update the resource.
 	// Do not attempt to retry if the resource version is specified.
 	if rv != "" {
+		// Clean out the resource version to always get the latest revision.
+		resource.(ResourceObject).GetObjectMeta().SetResourceVersion("")
 		// Validate the metadata is not changed for the the resource.
 		ro, err := rh.get(ctx, client, resource)
 		if err != nil {
 			return ro, err
 		}
-		// Set the specific resource version we are attempting to update.
-		ro.GetObjectMeta().SetResourceVersion(rv)
+		// Check that the resource version is the latest
+		if rv != ro.GetObjectMeta().GetResourceVersion() {
+			return ro, fmt.Errorf("Resource version '%s' is out of date (latest: %s). Update the resource in order to make changes.", rv, ro.GetObjectMeta().GetResourceVersion())
+		}
```

Comment:
To ensure you get the same error messages, I think you either want to:
-  return a ErrorResourceUpdateConflict type here
or
-  set the resource version in the merged settings and let the update return the error message as appropriate.

Reply:
As an aside - I'm not sure what you mean by "Update the resource" in the error - I assume you mean "update the YAML/JSON".  Anyways, if you return the correct error type you don't need to worry about crafting the correct message.

Reply:
I was hoping to avoid the error returned by the update since in the case of etcd compaction, it gives an unhelpful error message back about compaction which isn't the real issue the user should fix so I think using an `ErrorResourceUpdateConflict` might be the option I go with. 

Question about `ErrorResourceUpdateConflict`: is a mesage of "update conflict: <resource identifier>" enough to point out to a user? Is there a way that I could use `ErrorResourceUpdateConflict` other than just putting in a small blurb about resource version in the <identifier> portion of the message? It will work according to the code, it just doesn't look like the way the code was meant to be used...

Reply:
I think it's fine - basically an old resource version means that someone has modified the resource since you took your snapshot.